### PR TITLE
#Issue 3 - Attempt to add support for ENV

### PR DESCRIPTION
**Pull request by actimetal96**
[26/11/2019 - Diogo Azevedo]

Attempt to add support for "env".

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,6 +14,23 @@ module.exports = plugins => ({
       return config
     }, config)
   },
+// 26/11/2019 - Diogo Azevedo
```
Comment (satazor) [inline index.js]: It's not usual to have the date and your name in code.

File: index.js [hunk 2]
```diff
@@ -14,6 +14,23 @@ module.exports = plugins => ({
       return config
     }, config)
   },
+// 26/11/2019 - Diogo Azevedo  
+//added support for env
```
Comment (satazor) [inline index.js]: This comment seems unnecessary to me.

File: index.js (hunk 1 above)
Comment (satazor) [inline index.js]: It's not usual to have the date and your name in code.

File: index.js [hunk 3]
```diff
@@ -14,6 +14,23 @@ module.exports = plugins => ({
       return config
     }, config)
   },
+// 26/11/2019 - Diogo Azevedo  
+//added support for env 
+  env(config){
```
Comment (satazor) [inline index.js]: To keep consistency with the rest of the code, add a space before the brackets.

File: index.js [hunk 4]
```diff
@@ -14,6 +14,23 @@ module.exports = plugins => ({
...
+    return plugins.reduce((config, plugin) => {
+      if (plugin instanceof Array) {
+        const [_plugin, ...args] = plugin
+        plugin = _plugin(...args)
+      }
+      if (plugin instanceof Function) {
+        plugin = plugin()
+      }
+      if (plugin && plugin.env instanceof Function) {
```
Comment (satazor) [inline index.js:28]: I think `env` is an object that we should mixin with all the others. At least that's what I interpreted from #3

**Comment by satazor** [comment]
You forgot to update the documentation mentioning we can now have `env`.

Moreover, please reference the issue in the commit body like so: "Closes #3"

**Review by satazor — CHANGES_REQUESTED** [review]

**Comment by actimetal96** [comment]
Thank you for your feedback @satazor ,  working on a proprietary solution right now, it is common to leave your name/date when some major fix or feature is implemented, I guess some old habits are bad habits! Also, I didn't add documentation because I was not sure that my changes would add support for the "env", then I wouldn't write something that the project could do, but in reality, it couldn't.
I am going to fix everything regarding your feedback.
Thanks again

**Comment by JerryCauser** [comment]
Sorry, but `env` is just field in the `next.config.js`. There is no reason in that code here. Just use spread operator or Object.assign to insert static fields in result config.

**Pull request closed without merging**